Buddies.Web/src/pages/leaderboard.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import Avatar from '@mui/material/Avatar';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import { Box, Button, Container } from '@mui/material';
+import EmailIcon from '@mui/icons-material/Email';
+import TablePagination from '@mui/material/TablePagination';
+import { useEffect } from 'react';
+import CardActionArea from '@mui/material/CardActionArea';
+import { getLeaderboard } from '../api';
+
+type User = {
+  firstName: string,
+  lastName: string,
+  email: string,
+  userId: number,
+  buddyScore: number
+};
+
+let totalUsers: number = 0;
+
+/* Leaderboard page. Responsible for putting all the components that make up the
+  page together. It also sends GET requests to get all the users with the top
+  buddy score
+*/
+const Leaderboard = () => {
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+  const [leaderboard, setLeaderboard] = React.useState<User[]>([]);
+
+  /* Gets leaderboard info and then
+     creates necessary global vars  and states.
+  */
+  function getAndMakeLeaderboard() {
+    getLeaderboard(page + 1, rowsPerPage).then((res) => {
+      totalUsers = res.data.totalPages * rowsPerPage;
+      setLeaderboard(res.data.users);
+    }).catch((error) => {
+      alert(error);
```

Comment:
Perhaps show a toast message indicating that the leaderboard could not be fetched.